private/buf/bufcli/errors.go

```diff
@@ -164,19 +169,29 @@ func NewTemplateNotFoundError(owner string, name string) error {
 // Note that this function will wrap the error so that the underlying error
 // can be recovered via 'errors.Is'.
 func wrapError(err error) error {
-	if err == nil || (err.Error() == "" && !rpc.IsError(err)) {
-		// If the error is nil or empty and not an rpc error, we return it as-is.
-		// This is especially relevant for commands like lint and breaking.
+	// Attempt to convert err to a Connect error.  If it cannot be converted, we return it as-is.
+	// This is especially relevant for commands like lint and breaking.
+	connectError, ok := asConnectError(err)
+	if !ok {
 		return err
 	}
-	rpcCode := rpc.GetErrorCode(err)
+
+	connectCode := connectError.Code()
+	connectMessage := connectError.Message()
 	switch {
-	case rpcCode == rpc.ErrorCodeUnauthenticated, isEmptyUnknownError(err):
+	case connectCode == connect.CodeUnauthenticated, isEmptyUnknownError(err):
 		return errors.New(`Failure: you are not authenticated. Create a new entry in your netrc, using a Buf API Key as the password. For details, visit https://docs.buf.build/bsr/authentication`)
-	case rpcCode == rpc.ErrorCodeUnavailable:
+	case connectCode == connect.CodeUnavailable:
 		return fmt.Errorf(`Failure: the server hosted at that remote is unavailable: %w`, err)
 	}
-	return fmt.Errorf("Failure: %w", err)
+	return fmt.Errorf("Failure: %s", connectMessage)
```

Comment:
Although this could have happened before if `err.Error() == "" && rpc.IsError(err)`, should we consider an explicit check for `connectMessage == ""` and output `Failure: unknown error` instead of `Failure: ` .. technically this should never happen.

There is a `isEmptyUnknownError` function, but it only catches empty strings on "Unknown" errors.

Reply:
Actually this can happen if you do something like `buf beta registry repository list buf.build/sayers` (i.e. try to list repositories using an address that already has a repository in it).

In this case, Connect seems to return `Unimplemented` with no message, so your `Failure: ` scenario will occur.  I'll think of a better way to handle that.

Reply:
For posterity, this is now being handled in the CLI itself by validating the subset of commands that do not expect subdirectories.  Connect is behaving as it should with its returned codes.